src/puya/ir/builder/main.py

```diff
@@ -394,6 +409,15 @@ def visit_string_constant(self, expr: awst_nodes.StringConstant) -> BytesConstan
             value = None
         if value is None or len(value) > algo_constants.MAX_BYTES_LENGTH:
             raise CodeError(f"invalid {expr.wtype} value", expr.source_location)
+        match expr.wtype:
+            case wtypes.string_wtype:
+                pass
+            case wtypes.arc4_string_alias:
+                value = len(value).to_bytes(2) + value
```

Comment:
Again be nice to keep ARC4 stuff together, I think there is a function in arc4 module for encoding the length header, maybe use it here?

Also move the value length check after the match block to cover ARC4 values

Reply:
I had a quick look at the arc4 builder stuff and any existing code there is aimed at writing the IR to do the encoding rather than encoding it at compile time. We could add something to do the encoding but 90% of it would look the same as this function which feels a little redundant to duplicate